### ISSUE TITLE
How to make cmake replies on the another make rule?

### ISSUE THREAD

**Issue by yijianli-autra** [bazelbuild/rules_foreign_cc#972]
I am trying to build https://github.com/aws/aws-sdk-cpp with this rule.
I can build it with a build file like
```
load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")

package(default_visibility = ["//visibility:public"])

licenses(["notice"])

filegroup(
    name = "aws_cpp_sdk_files",
    srcs = glob(
        ["**"],
        exclude = ["crt/aws-crt-cpp/crt/aws-c-common/tests/**"],
    ),
)

cmake(
    name = "aws_s3_sdk",
    cache_entries = {
        "CMAKE_BUILD_TYPE": "Release",
        "BUILD_ONLY": "s3",
        "BUILD_SHARED_LIBS": "ON",
        "ENABLE_TESTING": "OFF",
    },
    copts = [
        "-Wno-sign-compare",  # There is a sign-compare warning in the aws sdk
    ],
    lib_source = ":aws_cpp_sdk_files",
    out_shared_libs = [
        "libaws-cpp-sdk-core.so",
        "libaws-cpp-sdk-s3.so",
    ],
)
```
*Note*, the system has installed the libcurl, because the aws-sdk-cpp relies on it.

But in our cases, we want to include the curl into our build chain. So I added another rule for curl
```
load("@rules_foreign_cc//foreign_cc:defs.bzl", "configure_make")

package(default_visibility = ["//visibility:public"])

licenses(["notice"])

filegroup(
    name = "curl_files",
    srcs = glob(["**"]),
)

configure_make(
    name = "curl",
    configure_in_place = True,
    lib_source = ":curl_files",
    out_include_dir = "include",
    out_static_libs = [
        "libcurl.a",
    ],
    out_shared_libs = [
        "libcurl.so",
    ],
    configure_options = [
        "--with-openssl", 
    ],
    targets = [
        "",
        "install",
    ],
)
```
It still works fine. But i am confusing about how to include it into the aws-sdk-cpp.
```
load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")

package(default_visibility = ["//visibility:public"])

licenses(["notice"])

filegroup(
    name = "aws_cpp_sdk_files",
    srcs = glob(
        ["**"],
        exclude = ["crt/aws-crt-cpp/crt/aws-c-common/tests/**"],
    ),
)

cmake(
    name = "aws_s3_sdk",
    cache_entries = {
        "CMAKE_BUILD_TYPE": "Release",
        "BUILD_ONLY": "s3",
        "BUILD_SHARED_LIBS": "ON",
        "ENABLE_TESTING": "OFF",
        "CURL_LIBRARY": "external/curl/curl/lib/libcurl.a", (confusing)
        "CURL_INCLUDE_DIR": "external/curl/curl/include", (confusing)
    },
    copts = [
        "-Wno-sign-compare",  # There is a sign-compare warning in the aws sdk
    ],
    lib_source = ":aws_cpp_sdk_files",
    out_shared_libs = [
        "libaws-cpp-sdk-core.so",
        "libaws-cpp-sdk-s3.so",
    ],
    deps = [
        "@curl//:curl", (confusing)
    ]
)
```
it did not work, since i pass the CURL_LIBRARY and CURL_INCLUDE_DIR. It could not find the right position in this case.  How can i get the right output from the curl?

**Comment by yijianli-autra** [comment]
resolved

**Comment by Ryang20718** [comment]
@yijianli-autra wondering how you resolved it?
